### ISSUE TITLE
Update CDP's accumulated fees and fee's updated time on query CDP

### ISSUE THREAD

**Issue by denalimarsh** [Kava-Labs/kava#366]
Accumulated fees are only updated on CDP actions. For CDP queries, the accumulated fees are calculated in memory and used to calculate the collateralization ratio, but the state of the CDP is not updated. Therefore, a CDP query result will almost always have a collateralization ratio that doesn't match the accumulated fees:
```
- cdp:
    id: 1
    owner: kava1xy7hrjy9r0algz9w3gzm8u6mrpq97kwta747gj
    collateral:
    - denom: xrp
      amount: "80000000"
    principal:
    - denom: usdx
      amount: "10000000"
    accumulated_fees: []
    fees_updated: 2020-02-08T17:07:16.233114Z
  collateral_value:
    denom: usdx
    amount: "20000000"
  collateralization_ratio: "1.999997200003919995"
```

We could insert the accumulated fees into the AugmentedCDP query result object _or_ update CDP state before returning the result. Does a CLI query updating state go against best practices?

**Comment by denalimarsh** [comment]
We should run some tests on the adverse impacts of calculating fees for each CDP every block. Do we have metrics to inform the theory that these calculations are too heavy to do each block?

**Comment by karzak** [comment]
Not in favor of updating state for free in response to queries, sounds like a DOS vector. Am in favor of a passive strategy for updating the fees of risky CDPs. My initial proposal for this issue was:

In begin block, before liquidating cdps, query for cdps within some small percent of liquidation (2-5%) and update fees for them. 

In general, I think iterating over CDPs every block is doable up to some pretty large number, but I would want to know experimentally how performance scales before adopting that approach.

**Comment by denalimarsh** [comment]
A CDP could be within the 2-5% liquidation range when its fees are included, but the query won't return them. We've added a buffer, but still have the same problem.

What if at time of CDP creation we calculate the block at which the fees will put the CDP under the collateral asset's liquidation ratio (or within _x%_ of it)? We store in a 'projected liquidation' index and all CDPs within _n_ blocks of their projected liquidation block have their fees updated. The CDP's projected liquidation block would be recalculated on changes to CDP collateral/principal, an update to a collateral asset's stability fee, or changes in collateral price greater than _x/2%_.

Also, I'm 200% in favor of setting up some performance tests and using them as a benchmark for this update (and future features).

**Comment by karzak** [comment]
Closed via #449
